### DISCUSSION TITLE
Redirect rule causes error `redirected you too many times`

### DISCUSSION THREAD

**Discussion by DonHuskini** [vercel/next.js#43912] [Help]
I want to redirect the exact path `/checkout` to `/checkout#addon`. All other paths should not be redirected. `/checkout#warranty` for example, should not be redirected.

I tried this solution and it leads to the error `redirected too many times` what makes sense because the path `/checkout#addon` is the same as `/checkout`. Can you help me to figure this out?
```
module.exports = {
  async redirects() {
    return [
      {
        source: '/checkout',
        destination: '/checkout#addons',
        permanent: true,
      },
    ];
  },
};
```

**Comment by icyJoseph** [comment]
I think the problem is that the server sends `/checkout#addons` as location header, the browser follows it, and makes a request for it to the server, but browsers never share the fragment (`#addons`), with the server. This is by design of the HTTP spec AFAIK.

So your server sees a new request for `checkout` and does the redirect once again. 

Read more here, https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#anchor:

> It is worth noting that the part after the #, also known as the fragment identifier, is never sent to the server with the request.
